Compiling >=1.44.0 with Angular causes error

**Describe the bug**
I wanted to build an Angular project with force-graph. However, I got this error: 

```
✘ [ERROR] TS2344: Type 'NodeType' does not satisfy the constraint 'NodeObject'. [plugin angular-compiler]

    node_modules/force-graph/src/index.d.ts:193:155:
      193 │ ...ance<ForceGraphInstance<NodeType, LinkType>, NodeType, LinkType>;
          ╵                                                 ~~~~~~~~

  This type parameter might need an `extends NodeObject` constraint.

    node_modules/force-graph/src/index.d.ts:193:31:
      193 │ ...e ForceGraphInstance<NodeType = NodeObject, LinkType = LinkObj...
          ╵                         ~~~~~~~~~~~~~~~~~~~~~
```

**Additional context**
Contents of my `package.json` (only relevant dependencies included):

```json
...
  "dependencies": {
    "@angular-eslint/eslint-plugin": "^18.4.2",
    "@angular/animations": "^19.0.1",
    "@angular/cdk": "^19.0.1",
    "@angular/common": "^19.0.1",
    "@angular/compiler": "^19.0.1",
    "@angular/core": "^19.0.1",
    "@angular/forms": "^19.0.1",
    "@angular/material": "^19.0.1",
    "@angular/platform-browser": "^19.0.1",
    "@angular/platform-browser-dynamic": "^19.0.1",
    "@angular/router": "^19.0.1",
    "@angular/service-worker": "^19.0.1",
    "@schematics/angular": "^19.0.2",
    "d3": "^7.8.5",
    "force-graph": "^1.44.0",
    "jszip": "^3.10.1",
    "tslib": "^2.5.0",
    "zone.js": "~0.15.0"
  },
  "devDependencies": {
    "@angular-devkit/build-angular": "^19.0.2",
    "@angular-eslint/builder": "18.4.2",
    "@angular-eslint/eslint-plugin-template": "18.4.2",
    "@angular-eslint/schematics": "18.4.2",
    "@angular-eslint/template-parser": "18.4.2",
    "@angular/cli": "^19.0.2",
    "@angular/compiler-cli": "^19.0.1",
    "@types/d3": "^7.4.0",
    "@types/node": "^22.10.0",
    "@typescript-eslint/eslint-plugin": "^7.2.0",
    "@typescript-eslint/parser": "^7.2.0",
    "eslint": "^8.57.0",
    "typescript": "5.6.3"
  },
  "engines": {
    "npm": ">=10.9.1",
    "node": ">=22.11.0"
  }
```

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -1,4 +1,4 @@
-export interface GraphData<N = NodeObject, L = LinkObject<N>> {
+export interface GraphData<N extends NodeObject = NodeObject, L extends LinkObject<N> = LinkObject<N>> {
   nodes: N[];
   links: L[];
 }
@@ -14,7 +14,7 @@ export interface NodeObject {
   fy?: number;
 }
 
-export interface LinkObject<N = NodeObject> {
+export interface LinkObject<N extends NodeObject = NodeObject> {
   source?: string | number | N;
   target?: string | number | N;
 }
@@ -30,7 +30,7 @@ export type CanvasPointerAreaPaintFn<T> = (obj: T, paintColor: string, canvasCon
 
 type DagMode = 'td' | 'bu' | 'lr' | 'rl' | 'radialout' | 'radialin';
 
-interface ForceFn<N = NodeObject> {
+interface ForceFn<N extends NodeObject = NodeObject> {
   (alpha: number): void;
   initialize?: (nodes: N[], ...args: any[]) => void;
   [key: string]: any;
@@ -190,6 +190,6 @@ export declare class ForceGraphGeneric<ChainableInstance, N extends NodeObject =
   graph2ScreenCoords(x: number, y: number): { x: number, y: number };
 }
 
-declare class ForceGraph<NodeType = NodeObject, LinkType = LinkObject<NodeType>> extends ForceGraphGeneric<ForceGraph<NodeType, LinkType>, NodeType, LinkType> {}
+declare class ForceGraph<N extends NodeObject = NodeObject, L extends LinkObject<N> = LinkObject<N>> extends ForceGraphGeneric<ForceGraph<N, L>, N, L> {}
 
 export default ForceGraph;

